Sources/SpeziHealthKit/Queries/HealthKitQuery.swift

```diff
@@ -0,0 +1,150 @@
+            anchor: anchor.hkAnchor,
+            limit: limit
+        )
+        let result = try await queryDescriptor.result(for: healthStore)
+        anchor = QueryAnchor(result.newAnchor)
+        return (added: result.addedSamples, deleted: result.deletedObjects)
+    }
+}
+
+
+extension HealthKit {
+    /// An element produced by continuous HealthKit queries to inform about updates to the HealthKit database
+    public struct ContinuousQueryElement<Sample: _HKSampleWithSampleType>: @unchecked Sendable {
+        // ^^SAFETY: this is in fact safe, since all of the update's (i.e., the `HKAnchoredObjectQueryDescriptor<Sample>.Result` type's)
+        // properties (i.e., deletedObjects, addedSamples, and newAnchor) are themselves Sendable. (rdar://16358485)
+        
+        typealias Update = HKAnchoredObjectQueryDescriptor<Sample>.Results.Element
+        
+        private let update: Update
+        
+        /// The samples which have been added since the last update.
+        public var addedSamples: [Sample] {
+            update.addedSamples
+        }
+        /// The objects which have been deleted since the last update.
+        public var deletedObjects: [HKDeletedObject] {
+            update.deletedObjects
+        }
+        /// The new query anchor, representing the state of the database as of directly after this update.
+        public var newAnchor: QueryAnchor<Sample> {
+            QueryAnchor(update.newAnchor)
+        }
+        
+        fileprivate init(update: Update) {
+            self.update = update
+        }
+    }
+    
+    
+    /// Runs a long-running query of HealthKit data.
+    ///
+    /// Use this function to run a continuous, long-running HealthKit data query.
+    /// This function returns an `AsyncSequence`, which will emit new elements whenever HealthKit informs us about changes to the database.
+    ///
+    /// - parameter sampleType: The ``SampleType`` that should be queried for.
+    /// - parameter timeRange: The time range for which the query should return samples.
+    /// - parameter anchor: A ``QueryAnchor``, which allows the caller to run a query that fetches only those objects which have been added since the last time the query was run.
+    /// - parameter limit: The maximum number of samples the query will return.
+    /// - parameter filterPredicate: Optional refining predicate that allows you to filter which samples should be fetched.
+    public func continuousQuery<Sample>(
+        _ sampleType: SampleType<Sample>,
+        timeRange: HealthKitQueryTimeRange,
+        anchor: QueryAnchor<Sample>,
+        limit: Int? = nil,
+        predicate filterPredicate: NSPredicate? = nil
+    ) -> AsyncMapSequence<HKAnchoredObjectQueryDescriptor<Sample>.Results, ContinuousQueryElement<Sample>> {
+        // ^^^ we sadly can't use `some AsyncSequence<ContinuousQueryElement<Sample>, any Error>` here,
+        // since the Failure parameter is only available starting with iOS 18...
```

Comment:
What about making this function iOS18 only?

Or if that's not feasible, at least have an overload that is available for older versions but deprecated for iOS 18 and newer and then points to the function signature that has a proper failure parameter in place? We will have to make this a breaking release anyways.

Reply:
Sounds great; I would aim for the overload approach and pushing people to the better version for iOS 18.

Reply:
update: we can do this with an overload (though we'd need 2 additional functions to make this work).

to be honest, in my opinion having the explicit type instead of the `some` type is fine here. it behaves the same way as the some type, and it keeps our implementaton a lot cleaner than doing the overload stuff

Reply:
@PSchmiedmayer i implemented and pushed the overload-based workaround. let me know what you think